peers/peerpool.go

```diff
@@ -164,28 +162,29 @@ func (p *PeerPool) startDiscovery(server *p2p.Server) error {
 	return nil
 }
 
-func (p *PeerPool) stopDiscovery(server *p2p.Server) {
-	if server.DiscV5 == nil {
+func (p *PeerPool) stopDiscovery(discovery Discovery) {
+	if !discovery.Running() {
 		return
 	}
 
 	for _, t := range p.topics {
 		t.StopSearch()
 	}
+	if err := discovery.Stop(); err != nil {
+		log.Debug("discovery erred when was closed", "err", err)
```

Comment:
s/erred/errored/g

Reply:
i actually thought that this is a valid shortcut, np will change it